src/test/java/SearchingMethodsTest.java

```diff
@@ -0,0 +1,245 @@
+    }
+
+    @Test
+    void recursiveLinearSearch_targetAtArbitraryIndex_returnsCorrectIndex() {
+        Integer[] myArray = {0, 4, 8, 12, 18, 25, 31, 33, 36, 35, 37, 40, 40, 40, 49, 55, 61, 62, 63, 74, 81, 89, 90, 98};
+        Integer myTarget = 55;
+        assertEquals(15, SearchingMethods.recursiveLinearSearch(myTarget, myArray, 0));
+    }
+
+    @Test
+    void recursiveLinearSearch_multipleEqualTargets_returnsFirstOccurrenceIndex() {
+        Integer[] myArray = {0, 4, 8, 12, 18, 25, 31, 33, 36, 35, 37, 40, 40, 40, 49, 55, 61, 62, 63, 74, 81, 89, 90, 98};
+        Integer myTarget = 40;
+        assertEquals(11, SearchingMethods.recursiveLinearSearch(myTarget, myArray, 0));
+    }
+
+    @Test
+    void iterativeBinarySearch_nullArray_throwsException() {
+        Integer[] myArray = null;
+        Integer myTarget = 99;
+        assertThrows(NullPointerException.class, () -> SearchingMethods.iterativeBinarySearch(myTarget, myArray));
+    }
+
+    @Test
+    void iterativeBinarySearch_nullTarget_throwsException() {
+        Integer[] myArray = new Integer[99];
+        Integer myTarget = null;
+        assertThrows(NullPointerException.class, () -> SearchingMethods.iterativeBinarySearch(myTarget, myArray));
+    }
+
+    @Test
+    void iterativeBinarySearch_sizeZeroArray_returnsSentinel() {
+        Integer[] myArray = new Integer[0];
+        Integer myTarget = 99;
+        assertEquals(-1, SearchingMethods.iterativeBinarySearch(myTarget, myArray));
+    }
+
+    @Test
+    void iterativeBinarySearch_targetNotWithinArray_returnsSentinel() {
+        Integer[] myArray = {0, 4, 8, 12, 18, 25, 31, 33, 36, 35, 37, 40, 40, 40, 49, 55, 61, 62, 63, 74, 81, 89, 90, 98};
+        Integer myTarget = 99;
+        assertEquals(-1, SearchingMethods.iterativeBinarySearch(myTarget, myArray));
+    }
+
+    @Test
+    void iterativeBinarySearch_targetAtZeroIndex_returnsCorrectIndex() {
+        Integer[] myArray = {0, 4, 8, 12, 18, 25, 31, 33, 36, 35, 37, 40, 40, 40, 49, 55, 61, 62, 63, 74, 81, 89, 90, 98};
+        Integer myTarget = 0;
+        assertEquals(0, SearchingMethods.iterativeBinarySearch(myTarget, myArray));
+    }
+
+    @Test
+    void iterativeBinarySearch_targetAtNthIndex_returnsCorrectIndex() {
+        Integer[] myArray = {0, 4, 8, 12, 18, 25, 31, 33, 36, 35, 37, 40, 40, 40, 49, 55, 61, 62, 63, 74, 81, 89, 90, 98};
+        Integer myTarget = 98;
+        assertEquals(23, SearchingMethods.iterativeBinarySearch(myTarget, myArray));
+    }
+
+    @Test
+    void iterativeBinarySearch_targetAtArbitraryIndex_returnsCorrectIndex() {
+        Integer[] myArray = {0, 4, 8, 12, 18, 25, 31, 33, 36, 35, 37, 40, 40, 40, 49, 55, 61, 62, 63, 74, 81, 89, 90, 98};
+        Integer myTarget = 55;
+        assertEquals(15, SearchingMethods.iterativeBinarySearch(myTarget, myArray));
+    }
+
+    @Test
+    void iterativeBinarySearch_unorderedArray_returnsSentinel() {
```

Comment:
That's not the defined behaviour for binary search in general. Searching an unordered collection may still yield a result (consider the list [3,2,1,3] and search for 3, you'll touch indexes 1, 2 then 3 and correctly return 3).

Reply:
I did take pause on this one as, well, exactly what you said. Easiest to just remove? :bomb:

Reply:
Imo, yes. Leaving tests out is a way to show that behaviour is undefined.